src/main/java/org/mifos/processor/bulk/API/implementation/BatchTransactionsController.java

```diff
@@ -33,37 +52,76 @@ public class BatchTransactionsController implements BatchTransactions {
     @Autowired
     FileStorageService fileStorageService;
 
+    @Autowired
+    RestRequestConvertor restRequestConvertor;
+
+    @Value("#{'${tenants}'.split(',')}")
+    protected List<String> tenants;
+    @Autowired
+    private CsvMapper csvMapper;
+
     @SneakyThrows
     @Override
-    public String batchTransactions(HttpServletResponse httpServletResponse,
-                                    String requestId, MultipartFile file, String fileName,
-                                    String purpose, String type, String tenant) throws IOException {
+    public String batchTransactions(
+            HttpServletRequest httpServletRequest,
+            HttpServletResponse httpServletResponse,
+            String requestId,
+            String fileName,
+            String purpose,
+            String type,
+            String tenant,
+            String registeringInstitutionId,
+            String programId) {
+
         log.info("Inside api logic");
-        String localFileName = fileStorageService.save(file);
-        Headers headers = new Headers.HeaderBuilder()
-                .addHeader("X-CorrelationID", requestId)
-                .addHeader(PURPOSE,purpose)
-                .addHeader(FILE_NAME,localFileName)
-                .addHeader("Type",type)
-                .addHeader("Platform-TenantId",tenant)
-                .build();
-
-        Exchange exchange = SpringWrapperUtil.getDefaultWrappedExchange(producerTemplate.getCamelContext(),
-              headers);
-        exchange = producerTemplate.send("direct:post-batch-transactions", exchange);
-        int statusCode = exchange.getIn().getHeader(Exchange.HTTP_RESPONSE_CODE, Integer.class);
-		httpServletResponse.setStatus(statusCode);
-       	return exchange.getIn().getBody(String.class);
+        Headers.HeaderBuilder headerBuilder = new Headers.HeaderBuilder()
+                .addHeader(HEADER_CLIENT_CORRELATION_ID, requestId)
+                .addHeader(PURPOSE, purpose)
+                .addHeader(HEADER_TYPE, type)
+                .addHeader(HEADER_PLATFORM_TENANT_ID, tenant)
+                .addHeader(HEADER_REGISTERING_INSTITUTE_ID, registeringInstitutionId)
+                .addHeader(HEADER_PROGRAM_ID, programId);
+
+        if (JWSUtil.isMultipartRequest(httpServletRequest)) {
```

Comment:
because that is passed by user and in postman usually the content-type is autopopulated, and is more of standard as per REST. So thats why I thought of using "CONTENT-TYPE" header to judge if its a multipart request or json request. 
Also JWSUtil is implemented in connector-common and hence it serve the purpose of finding the multipart request irrespective of the custom headers, in this case "raw"/"csv".

Reply:
As per design, the type parameter is to be passed explicitly. The approach you have used is good. yet we have to hold the API documentation and design. So add a validation to check whether the type parameter matches the content type or not.

Reply:
added the header validation